### PR TITLE
Add support for stricter edition links

### PR DESCRIPTION
**Pull request by thomasleese**
This way we can use a white list rather than a black list.

### CODE REVIEW

File: lib/govuk_content_schemas/schema_combiner.rb [hunk 1]
```diff
@@ -53,11 +53,18 @@ def add_details(schema)
...
     linkset_links = schemas[:links]
+    base_edition_links = schemas[:base_edition_links]
+    edition_links = schemas[:edition_links]
+
+    order = [
+      edition_links,
+      linkset_links,
+      base_edition_links,
+      base_linkset_links
+    ].compact
```
Comment (carlosmartinez) [inline lib/govuk_content_schemas/schema_combiner.rb]: could this be replaced with

```
order = schemas.slice(:edition_links, :links, :base_edition_links, :base_links).compact
```

Comment (thomasleese) [inline lib/govuk_content_schemas/schema_combiner.rb]: I don't think we can do that as it won't preserve the order when turning it into an array.

Comment (carlosmartinez) [inline lib/govuk_content_schemas/schema_combiner.rb]: ah, right. could require ActiveSupport core extensions and do `slice.values`, but probably not worth it.

File: lib/tasks/combine_schemas.rake [hunk 2]
```diff
@@ -30,7 +30,7 @@ end
 
 def sources_for_v2_schema(filename)
   Rake::FileList.new(
-    "formats/definitions.json",
+    "formats/{definitions,base_edition_links}.json",
```
Comment (carlosmartinez) [inline lib/tasks/combine_schemas.rake]: what

Comment (thomasleese) [inline lib/tasks/combine_schemas.rake]: To generate a v2 `schema.json` file, information from both `formats/definitions.json` and `formats/base_edition_links.json` is required.

File: lib/govuk_content_schemas/schema_combiner.rb [hunk 3]
```diff
@@ -53,20 +53,31 @@ def add_details(schema)
...
 
-  def merge_schemas(base_schema, other)
+  def merge_schemas(base_schema, *others)
     merged_schema = embeddable_schema(base_schema)
-    other = embeddable_schema(other)
-    merged_schema["properties"] = merged_schema["properties"].merge(other["properties"])
+
+    others
+      .map { |other| embeddable_schema(other) }
+      .each { |other| merged_schema["properties"].merge!(other["properties"]) }
```
Comment (carlosmartinez) [inline lib/govuk_content_schemas/schema_combiner.rb]: does the following work? i think it expresses the intent more clearly.

```
    other_properties = others.map do |other|
      embeddable_schema(other)["properties"]
    end
    merged_schema["properties"].merge!(other_properties))
```

Comment (carlosmartinez) [inline lib/govuk_content_schemas/schema_combiner.rb]: merged_schema["properties"].merge!(other_properties.reduce({}, :merge)))

**Pull request merged**